internal/business/server/http_server.go

```diff
@@ -3,24 +3,36 @@ package server
 import (
 	"context"
 	"errors"
+	"fmt"
 	"net"
 	"net/http"
 	"strings"
 
+	"github.com/openkcm/common-sdk/pkg/commoncfg"
 	"github.com/openkcm/common-sdk/pkg/fingerprint"
 	"github.com/samber/oops"
 
 	slogctx "github.com/veqryn/slog-context"
 
 	"github.com/openkcm/session-manager/internal/config"
-	"github.com/openkcm/session-manager/internal/middleware/responsewriter"
+	"github.com/openkcm/session-manager/internal/middleware"
 	"github.com/openkcm/session-manager/internal/openapi"
 	"github.com/openkcm/session-manager/internal/session"
 )
 
 // createStatusServer creates an API http server using the given config
-func createHTTPServer(_ context.Context, cfg *config.Config, sManager *session.Manager) *http.Server {
-	openAPIServer := newOpenAPIServer(sManager)
+func createHTTPServer(_ context.Context, cfg *config.Config, sManager *session.Manager) (*http.Server, error) {
+	csrfSecret, err := commoncfg.LoadValueFromSourceRef(cfg.SessionManager.CSRFSecret)
```

Comment:
We already load the secret in `NewManager` and there we also check the length. Would be nice to load it just once.